### PR TITLE
181283540 Add pagination to ping things

### PR DESCRIPTION
**Pull request by kbiala**
#### What's this PR do?
- Adds pagination to ping thing view

#### How should this be manually tested?
- Review in the browser

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/n/projects/2177859/stories/181283540)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: app/controllers/api/v1/ping_things_controller.rb [hunk 1]
```diff
@@ -11,14 +11,26 @@ class InvalidRecordCount < StandardError; end
       def index
         limit = [(index_params[:limit] || 240).to_i, 9999].min
         page = index_params[:page] || 1
+        with_total_count = index_params[:with_total_count].to_s == "true" ? true : false
```
Comment (maciejkocylapc) [inline app/controllers/api/v1/ping_things_controller.rb:14]:
```suggestion
        with_total_count = [true, "true"].include? index_params[:with_total_count]
```

File: app/javascript/packs/ping_thing/index_template.vue [hunk 2]
```diff
@@ -4,45 +4,80 @@
...
+        ping_things: [],
+        page: 1,
+        total_count: 0,
+        api_url: api_url
       }
     },
     created () {
       var ctx = this
-      axios.get("/api/v1/ping-thing/" + this.network)
+      axios.get(ctx.api_url)
```
Comment (maciejkocylapc) [inline app/javascript/packs/ping_thing/index_template.vue:53]: perhaps using axios params would be a better idea? 
`axios.get(ctx.api_url, { params: { with_total_count: true } })`

File: app/javascript/packs/ping_thing/index_template.vue [hunk 3]
```diff
@@ -4,45 +4,80 @@
...
+    watch: {
+      page: function(){
+        this.paginate()
+      }
+    },
+    methods: {
+      paginate: function(){
+        var ctx = this
+        var url = ctx.api_url + '&page=' + ctx.page
+        axios.get(url)
```
Comment (maciejkocylapc) [inline app/javascript/packs/ping_thing/index_template.vue:68]: like above

File: test/controllers/api/v1/ping_thing_controller_test.rb [hunk 4]
```diff
@@ -168,4 +168,40 @@ def setup
     json = response_to_json(@response.body)
     assert_equal 123, json.first["response_time"]
   end
+
+  test "GET api_v1_ping_things with with_total_count param, returns pigns and records count" do
```
Comment (maciejkocylapc) [inline test/controllers/api/v1/ping_thing_controller_test.rb:172]:
```suggestion
  test "GET api_v1_ping_things with with_total_count param, returns pings and records count" do
```

File: app/controllers/api/v1/ping_things_controller.rb (hunk 1 above)
Comment (kbiala) [reply app/controllers/api/v1/ping_things_controller.rb:14]: I cant's see how it's better. to_s will do exatly the same thing and having only one "true" in this statement is much easier to read imo.

**Pull request merged**